pkg/stub/handler.go

```diff
@@ -131,38 +187,74 @@ func (h *Handler) StoreCurrentValidConfig(condition v1alpha1.SamplesResourceCond
 		// 4.0 testing showed that we were getting empty ConfigMaps
 		cm = nil
 	}
-	if cm != nil {
-		// just return ... we only update the config map once
-		return nil
+	create := false
```

Comment:
It's a bit paranoid, but since you are fetching the configmap within this function, there's no guarantee the changes are valid anymore... (the configmap could have changed since you called SpecValidation).

I think this needs to be a continuous flow:

1) confirm the sampleresource is valid on its own (doesn't contain totally invalid values) - stop here if it is
2) fetch the configmap - if it does not exist, create one w/ the values from the SR
3) if the configmap exists, compare the changes, stop here and throw error if changes are unacceptable
4) assuming we didn't get an error earlier, proceed to apply the SR configuration:
   a) mutate the configmap you fetched in (3) to reflect the new configuration and apply/update it (if you get an error here because someone else changed the configmap underneath you, error out and retry)
   b) update/create the samples based on the new configuration
   
I also wonder if we can find our way to getting rid of this configmap entirely(not in this PR, but thinking ahead)...   the reason we introduced it was to ensure we had a copy of the previous config if someone deleted the SR.  But if we take the position that upon deletion of the SR, we delete all the samples, then we do not need to worry about that scenario.  If we did that, we could simply store the current config settings in Status on the SR itself, which I think makes a lot of this logic easier.

Reply:
(we may need to talk through 1-4 on a whiteboard, i suspect i've done an imperfect job explaining my thinking)

Reply:
Also, looked at the code to refresh my memory, and we *ALREADY ARE* deleting:
1. the imagestreams in openshift namespace, but factor in skipped filter
2. the templates in openshift namespace, but factor in skipped filter
3. the secret in openshift namespace
4. the config map in the operator's namespace

Since the operator does not create the secret in its namespace, we left it alone.

Reply:
OK @bparees and I decided to just stay pat on this thread for now, think more about it in the background.